### ISSUE TITLE
request gets hung up on sending

### ISSUE THREAD

**Issue by orebot** [jarrodek/ChromeRestClient#872]
Your description here

testing very basic api.  First get request works, then every one there after gets stuck on send

## Expected outcome
What should happen?

List of authors by ID should be shown

## Actual outcome
What happened?

Hung up on send, Abort button is shown with blue line scrolling/waiting

## Versions
App: 8.9.53.279-stable
Chrome: 54.0.2840.87
Platform: Windows (10)

## Steps to reproduce
1. enter get into url line
2. hit send, everything works
3. try to get another request, ARC hangs up, and never executes

**Comment by jarrodek** [comment]
Hi,

I can't reproduce the problem. Maybe the second endpoint is not working?
Also you can [debug](https://github.com/jarrodek/ChromeRestClient/wiki/debugging) the app and post error messages from the console for analysis.

**Comment by orebot** [comment]
Thank you very much.  I did the debug method, and I am gettin the error

"Uncaught TypeError:  Cannot read property 'length' of undefined(...)"
    push                                                   @elements.js:5168
    _processResponseWorkerData             @elements.js:121525


I'm new to api's and have no idea what this message means.  If you could
steer me the right direction, I would be very thankful

**Comment by f8k8** [comment]
I'm seeing the same issue. After the previous upgrade failed, I uninstalled and reinstalled fresh, and now there are issues with subsequent requests. The first request executes fine, but trying a second one just sits waiting (doesn't even time out). There are multiple errors shown in the debug console. Launching ARC, the console shows:

    chrome-extension://hgmloofddffdnphfgcellkdfbfbjeloo/bower_components/iron-flex-layout/classes/iron-shadow-flex-layout.js Failed to load resource: net::ERR_FILE_NOT_FOUND
    chrome-extension://hgmloofddffdnphfgcellkdfbfbjeloo/bower_components/iron-flex-layout/classes/iron-flex-layout.js Failed to load resource: net::ERR_FILE_NOT_FOUND
    
    index.html:1 window.localStorage is not available in packaged apps. Use chrome.storage.local instead.
    
    elements.js:13141 /deep/ combinator is deprecated. See https://www.chromestatus.com/features/6750456638341120 for more details.attached @ elements.js:13141
    
    elements.js:73327 Uncaught DOMException: Failed to execute 'objectStore' on 'IDBTransaction': The specified object store was not found.(…)
      addDeletedOrLocalIndex @ elements.js:73327
      next @ elements.js:74093
      req.onupgradeneeded @ elements.js:74097
      
    elements.js:69450 Failed to open indexedDB, are you in private browsing mode?
      guardedConsole @ elements.js:69450
      req.onerror @ elements.js:74194
    
    elements.js:120648 Fetch complete: POST "https://www.google-analytics.com/collect"
    elements.js:120648 Fetch complete: POST "https://www.google-analytics.com/collect"
    elements.js:120648 Fetch complete: POST "https://www.google-analytics.com/collect"
    
    blob:chrome-extension://hgmloofddffdnphfgcellkdfbfbjeloo/6cee312d-3297-4925-97a6-143f91dea450:18 Uncaught (in promise) DOMException: Failed to execute 'transaction' on 'IDBDatabase': One of the specified object stores was not found.(…)
      Promise @ blob:chrome-extension://hgmloofddffdnphfgcellkdfbfbjeloo/6cee312d-3297-4925-97a6-143f91dea450:18
      processor.queryKeys @ blob:chrome-extension://hgmloofddffdnphfgcellkdfbfbjeloo/6cee312d-3297-4925-97a6-143f91dea450:17
      processor.getDatabase.then @ blob:chrome-extension://hgmloofddffdnphfgcellkdfbfbjeloo/6cee312d-3297-4925-97a6-143f91dea450:65
      
At this point, ARC has started. Performing the first request (which returns successfully), shows the following in the console:
      
    elements.js:120648 Fetch complete: POST "https://www.google-analytics.com/collect"
    elements.js:120648 Fetch complete: POST "https://www.google-analytics.com/collect"
    elements.js:120648 Fetch complete: POST "https://www.google-analytics.com/collect"
    
    Uncaught TypeError: Cannot read property 'length' of undefined(…)
      push @ elements.js:5168
      _processResponseWorkerData @ elements.js:121529
    
After this, no subsequent requests will complete. When trying a second request, the console shows:
      
    elements.js:120648 Fetch complete: POST "https://www.google-analytics.com/collect"
    elements.js:120648 Fetch complete: POST "https://www.google-analytics.com/collect"
    elements.js:120648 Fetch complete: POST "https://www.google-analytics.com/collect"
    
If abort is pressed during one of the subsequent requests, the console shows the following error:
    
    elements.js:34196 Uncaught TypeError: this._mainPromise.reject is not a function(…)
      abort @ elements.js:34196
      abort @ elements.js:98115
      abortRequest @ elements.js:117539
      handler @ elements.js:3386
      fire @ elements.js:4152
      _abortRequest @ elements.js:91831
      handler @ elements.js:3386
      fire @ elements.js:4152
      fire @ elements.js:3724
      forward @ elements.js:4021
      click @ elements.js:4006
      handleNative @ elements.js:3614

**Comment by jarrodek** [comment]
Hi,

Thank you for the details. I found the bug and I'm working it right now. I'll update the app in CWS later when I'm sure that it's working.
I'm sorry for this bug. In one place array hasn't been initialized and the app is trying to push new result into not existing array which causes an error. It should work for the request (matching URL + method) that has been executed in the history after restarting the app because then this array should be initialized by other function.

**Comment by f8k8** [comment]
No problem - glad to help. I'll keep an eye out for the fix and give it a test when it's live. Thanks.

**Comment by jarrodek** [comment]
@f8k8 I've noticed something new in your console output. It looks like your database hasn't been initialised. I mean data stores hasn't been created at all.

Proper structure should look somewhat like this:
![image](https://cloud.githubusercontent.com/assets/3320020/20180730/94280cfe-a753-11e6-8c23-19d1bb700419.png)

Can you, using the debugger, check if you have this databases?

**Comment by f8k8** [comment]
`_pouch_history-data` is present, but empty. There's no arrow next to it to inspect it. Could this be something to do with the previous issue of upgrading (https://github.com/jarrodek/ChromeRestClient/issues/830)? To work around that issue, I deleted ARC and reinstalled from the Chrome Web Store.

![arc_db](https://cloud.githubusercontent.com/assets/1764665/20184081/5d608500-a75f-11e6-83b7-5ee9e786985d.png)

**Comment by jarrodek** [comment]
I'm not sure what this means. It might be an issue with PouchDB. Other possibility that there's some operation pending on this database and therefore it can't be accessed by other threads (IDB access is async and from the web worker).
I've added a test to the query function if the database exists. If not it will not try to read the database but it also disables assistant (new feature in the app).

I've created new issue for this (https://github.com/jarrodek/ChromeRestClient/issues/884) since it's not related to current issue.
